include/class.format.php

```diff
@@ -424,8 +424,9 @@ function($match) use ($exclude, $allowed) {
                 $m = array();
                 // Split the src URL and get the extension
                 preg_match('/src="([^"]+)"/', $match[2], $m);
-                $url = explode('.', explode('?', $m[1])[0]);
-                $ext = preg_split('/[^A-Za-z]/', end($url))[0];
+                $parts = parse_url($m[1]);
+                $path = explode('.', $parts['path']);
```

Comment:
You might want to account for a case where url is invalid - `parse_url` would return `false` in that case. I would also do the following.
```php
$patch = parse_url($url, PHP_URL_PATH);
```